services/galaxytoolshed/galaxy-toolshed-base.js

```diff
@@ -0,0 +1,56 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const orderedInstallableRevisionsSchema = Joi.array()
+  .items(Joi.string())
+  .required()
+
+export default class BaseGalaxyToolshedService extends BaseJsonService {
+  static defaultBadgeData = { label: 'galaxy-toolshed' }
+  static baseUrl = 'https://toolshed.g2.bx.psu.edu'
+
+  async fetchOrderedInstallableRevisionsSchema({ repository, owner }) {
+    return this._requestJson({
+      schema: orderedInstallableRevisionsSchema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_ordered_installable_revisions?name=${repository}&owner=${owner}`,
+    })
+  }
+
+  async fetchRepositoryRevisionInstallInfoSchema({
+    repository,
+    owner,
+    schema,
+    changesetRevision,
+  }) {
+    return this._requestJson({
+      schema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_repository_revision_install_info?name=${repository}&owner=${owner}&changeset_revision=${changesetRevision}`,
+    })
+  }
```

Comment:
Could you explain your thinking around why `schema` is an input parameter?

Something feels off here because there's typically a strict 1:1 relationship between an upstream endpoint we are calling, and the schema we use to validate/make assertions about the structure of the response provided by that upstream endpoint.

Additionally, I assume that because you'd placed this into the base class that it's a response we'll need for other, non-download/non-install related badges?

Reply:
Exactly, this function will be use by others utilities: 
* releases
* activity
* version
* ...